lib/ui/painting/image_descriptor.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_PAINTING_IMAGE_DESCRIPTOR_H_
+#define FLUTTER_LIB_UI_PAINTING_IMAGE_DESCRIPTOR_H_
+
+#include <cstdint>
+#include <memory>
+
+#include "flutter/fml/macros.h"
+#include "flutter/lib/ui/dart_wrapper.h"
+#include "flutter/lib/ui/painting/immutable_buffer.h"
+#include "third_party/skia/include/codec/SkCodec.h"
+#include "third_party/skia/include/core/SkImageInfo.h"
+#include "third_party/tonic/dart_library_natives.h"
+
+namespace flutter {
+
+class ImageDescriptor : public RefCountedDartWrappable<ImageDescriptor> {
+ public:
+  ~ImageDescriptor() override = default;
+
+  static void initEncoded(Dart_NativeArguments args);
+
+  static void initRaw(Dart_Handle descriptor_handle,
+                      fml::RefPtr<ImmutableBuffer> data,
+                      int width,
+                      int height,
+                      int row_bytes,
+                      int pixel_format);
+
+  void instantiateCodec(Dart_Handle callback,
+                        int target_width,
+                        int target_height);
+
+  int width() const { return image_info_.width(); }
+
+  int height() const { return image_info_.height(); }
+
+  int bytesPerPixel() const { return image_info_.bytesPerPixel(); }
+
+  void dispose() {
+    ClearDartWrapper();
+    codec_.reset();
+  }
+
+  size_t GetAllocationSize() const override {
+    return sizeof(ImageDescriptor) + sizeof(SkImageInfo) + sizeof(SkCodec);
```

Comment:
This is inaccurate. This object also owns the compressed image allocation. See my comment about this being more than just a descriptor.

Reply:
Also, this you are double counting`SkImageInfo` which is accounted for in `ImageDescriptor` already. This should be something like `return sizeof(decltype(*this)) + sizeof(SkCodec) + buffer_->getSize()`.

In any case, the other allocations are small, the buffer is the big one.